### ISSUE TITLE
Add ability to deny permission to user when permission belongs to a role

### ISSUE THREAD

**Issue by waylaidwanderer** [spatie/laravel-permission#82]
Let's say you create a permission as follows:

`Permission::create(['name' => 'chat']);`

Then you create this role and assign the permission:

`Role::create(['name' => 'regular'])->givePermissionTo('chat');`

Then you assign the role to the first user in your database:

`User::find(1)->assignRole('regular');`

Checking whether the user has the permission (`var_dump(User::find(1)->hasPermissionTo('chat'))`) gives the output `bool(true)`.

However, if you execute `User::find(1)->revokePermissionTo('chat');` and then check whether the user has the permission (`var_dump(User::find(1)->hasPermissionTo('chat'))`), it will still output `bool(true)`.

Basically could you add the ability to revoke a permission from a user (and only that specified user) when the permission belongs to a role?

**Comment by freekmurze** [comment]
It's best to approach permissions as additive, not subtractive.

If you don't want a user to have a certain permission just don't give that user a role that has that permission. Create a separate role without that permission.

**Comment by waylaidwanderer** [comment]
Let's say I have this role with the following permissions:

```
Role::create(['name' => 'regular'])->givePermissionTo('add funds')
                                    ->givePermissionTo('withdraw balance')
                                    ->givePermissionTo('chat')
                                    ->givePermissionTo('buy')
                                    ->givePermissionTo('deposit items')
                                    ->givePermissionTo('withdraw items');
```

I want the ability to revoke all these permissions modularly for a singular user. Is my only alternative to assign these permissions to every user that is created? It seems kind of excessive, and would be a lot of extra rows in the `user_has_permissions` table.

**Comment by freekmurze** [comment]
If the most common case is that users have every permission, I'd create an admin role and assign that to all users. If there's an exception or the user I'd revoke the role and start granting the individual permissions. What a few helper functions in your app this would not be that difficult to implement.

Seven different permissions is not that many. Maybe it's just simpler to create them separately and assign them all to every user.

I'd probably never accept a PR that adds subtractive permission behaviour to this package. So if you want that, you're of course free to fork this package and implement it yourself.

**Comment by waylaidwanderer** [comment]
Thanks for the reply, Freek. I appreciate your thoughts on this issue.
